common/netmap.go

```diff
@@ -12,6 +14,11 @@ import (
 // a successive call is not guaranteed.
 // Caller must have `NewEpoch` method with a single numeric argument.
 func SubscribeForNewEpoch() {
+	callingHash := runtime.GetExecutingScriptHash()
+	if !management.HasMethod(callingHash, "newEpoch", 1) {
```

Comment:
Why this check is here and not in the `subscribeForNewEpoch` itself? It makes much more sense there since `netmap` controls its state and shouldn't allow bad subscriptions (forget contract upgrades for a moment).